### PR TITLE
Extract AI support power handling to AISupportPowerManager

### PR DESCRIPTION
**Pull request by reaperrr**
Another step towards modularizing AI and making HackyAI.cs less monolithic.

### CODE REVIEW

File: OpenRA.Mods.Common/AI/HackyAI.cs [hunk 1]
```diff
@@ -280,6 +278,7 @@ public CPos GetRandomBaseCenter()
 		BitArray resourceTypeIndices;
 
 		AIHarvesterManager harvManager;
+		AISupportPowerManager supportPowerManager;
```
Comment (pchote) [inline OpenRA.Mods.Common/AI/HackyAI.cs]: Calling this `supportPowerManager` is asking for trouble.  This class already has a `SupportPowerManager supportPowerMngr`!

Can you please rename `supportPowerMngr` to `supportPowerManager` and then this to something else?

Comment (pchote) [reply OpenRA.Mods.Common/AI/HackyAI.cs]: Or, better, move `supportPowerMngr` inside the new class (but also fix the variable naming)

**Comment by reaperrr** [comment]
Updated. Moved FrozenActorLayer caching to AISupportPowerManager as well, since it's only used there.

File: OpenRA.Mods.Common/AI/HackyAI.cs [hunk 2]
```diff
@@ -336,11 +330,10 @@ public void Activate(Player p)
 			Player = p;
 			IsEnabled = true;
 			playerPower = p.PlayerActor.Trait<PowerManager>();
-			supportPowerMngr = p.PlayerActor.Trait<SupportPowerManager>();
 			playerResource = p.PlayerActor.Trait<PlayerResources>();
-			frozenLayer = p.PlayerActor.Trait<FrozenActorLayer>();
 
 			harvManager = new AIHarvesterManager(this, p);
```
Comment (pchote) [inline OpenRA.Mods.Common/AI/HackyAI.cs]: Would you mind renaming `aiHarvManager` too for consistency?  Or I guess we could go back to `supportPowerManager` as the ambiguity is now solved.

**Comment by reaperrr** [comment]
> Or I guess we could go back to supportPowerManager as the ambiguity is now solved.

Done, which is why the comments aren't hidden.

**Review by GraionDilach — APPROVED** [review]
:+1:

**Review by abcdefg30 — APPROVED** [review]

**Pull request merged**